apps/examples/testcase/le_tc/network/itc_net_connect.c

```diff
@@ -0,0 +1,253 @@
+	}
+	memset(&serv_addr, '0', sizeof(serv_addr));
+	serv_addr.sin_family = AF_INET;
+	serv_addr.sin_addr.s_addr = inet_addr("127.0.0.1");
+	serv_addr.sin_port = htons(g_port);
+	ret = connect(sock, (struct sockaddr *)&serv_addr, sizeof(serv_addr));
+	if (ret != OK) {
+		close(sock);
+		return NULL;
+	}
+	strncpy(client_msg, CLIENT_MSG, sizeof(CLIENT_MSG));
+	client_msg[strlen(client_msg) - 2] = '0' + (*id);
+	ret = send(sock, client_msg, strlen(client_msg), 0);
+	if (ret != strlen(client_msg)) {
+		close(sock);
+		return NULL;
+	}
+	valread = read(sock, buffer, BUFF_LEN);
+	if (valread != strlen(client_msg) + strlen(SERVER_MSG)) {
+		close(sock);
+		return NULL;
+	}
+	if (strncmp(buffer, client_msg, strlen(client_msg)) != 0) {
+		close(sock);
+		return NULL;
+	}
+	if (strncmp(buffer + strlen(client_msg), SERVER_MSG, strlen(SERVER_MSG)) != 0) {
+		close(sock);
+		return NULL;
+	}
+	sleep(WAIT_TIME_SEC_2);
+	ret = close(sock);
+	if (ret != OK) {
+		return NULL;
+	}
+	return NULL;
+}
+
+/**
+* @testcase         :itc_net_connect_p
+* @brief            :connects the socket referred to by the file descriptor
+* @scenario         :start server and client thread, connect client to server
+* @apicovered       :connect()
+* @precondition     :none
+* @postcondition    :none
+*/
+static void itc_net_connect_p(void)
+{
+	pthread_t server_thread;
+	pthread_t client_thread;
+	int num_clients = 1;
+	int id = 1;
+	int ret;
+	int *ptr = &num_clients;
+	int *ptr_id = &id;
+	g_port = 8080;
+
+	pthread_create(&server_thread, NULL, server_connect, ptr);
+	sleep(WAIT_TIME_SEC_2);
+	pthread_create(&client_thread, NULL, client_connect, ptr_id);
+
+	sleep(WAIT_TIME_SEC_3);
+
+	ret = pthread_join(server_thread, NULL);
+	TC_ASSERT_EQ_CLEANUP("pthread_join", ret, OK, pthread_join(client_thread, NULL));
+	ret = pthread_join(client_thread, NULL);
+	TC_ASSERT_EQ("pthread_join", ret, OK);
+
+	TC_SUCCESS_RESULT();
+}
+
+/**
+* @testcase         :itc_net_connect_p_multiple_clients
+* @brief            :connects the socket referred to by the file descriptor
+* @scenario         :start server and 3 client thread, connect client thread to server one by one
+* @apicovered       :connect()
+* @precondition     :none
+* @postcondition    :none
+*/
+static void itc_net_connect_p_multiple_clients(void)
+{
+	pthread_t server_thread;
+	pthread_t client1_thread;
+	pthread_t client2_thread;
+	pthread_t client3_thread;
+	int num_clients = 3;
+	int id1 = 1;
+	int id2 = 2;
+	int id3 = 3;
+	int ret;
+	int *ptr_num_clients = &num_clients;
+	int *ptr_id1 = &id1;
+	int *ptr_id2 = &id2;
+	int *ptr_id3 = &id3;
+	g_port = 8090;
+
+	pthread_create(&server_thread, NULL, server_connect, ptr_num_clients);
+	sleep(WAIT_TIME_SEC_2);
+	pthread_create(&client1_thread, NULL, client_connect, ptr_id1);
+	pthread_create(&client2_thread, NULL, client_connect, ptr_id2);
+	pthread_create(&client3_thread, NULL, client_connect, ptr_id3);
+
+	sleep(WAIT_TIME_SEC_7);
+
+	ret = pthread_join(server_thread, NULL);
```

Comment:
same above

Reply:
I can get result by passing "void ** "variable but since test case are using ASSERT which provide Pass/Fail condition; so I didn't validate thread return value.